sdk/python/feast/feature_view.py

```diff
@@ -207,9 +198,49 @@ def __init__(
             owner=owner,
         )
         self.online = online
-        self.stream_source = stream_source
         self.materialization_intervals = []
 
+    def _initialize_sources(self, name, batch_source, stream_source, source):
+        if source:
+            if (
+                isinstance(source, PushSource)
+                or isinstance(source, KafkaSource)
+                or isinstance(source, KinesisSource)
+            ):
+                self.stream_source = source
+                if not source.batch_source:
+                    raise ValueError(
+                        f"A batch_source needs to be specified for stream source `{source.name}`"
+                    )
+                else:
+                    self.batch_source = source.batch_source
+            else:
+                self.stream_source = stream_source
+                self.batch_source = source
+        else:
+            warnings.warn(
+                "batch_source and stream_source have been deprecated in favor or `source`."
+                "The deprecated fields will be removed in Feast 0.23.",
+                DeprecationWarning,
+            )
+            if stream_source is not None and isinstance(stream_source, PushSource):
+                if stream_source.batch_source is None or not isinstance(
+                    stream_source.batch_source, DataSource
+                ):
+                    raise ValueError(
```

Comment:
note: I _think_ this is an unnecessary check since we already check this during initialization of a `PushSource`? but I'm fine leaving it as is

Reply:
Okay was able to remove.